src/main/kotlin/io/github/mojira/arisa/ModuleExecutor.kt

```diff
@@ -0,0 +1,333 @@
+                        is FailedModuleResponse -> for (exception in it.exceptions) {
+                            log.error("[RESPONSE] [$issue] [${response.first}] Failed", exception)
+                        }
+                    }
+                }, {
+                    processedIssues[issue] = true
+                    log.info("[RESPONSE] [$issue] [${response.first}] Successful")
+                })
+            }
+    }
+
+    fun execute() {
+        // Cache issues returned by a query to avoid searching the same query for different modules
+        val queryCache = mutableMapOf<String, List<Issue>>()
+        val processedTickets = mutableMapOf<String, Boolean>()
+        val exec = ::executeModule.partially2(queryCache).partially2(processedTickets)
+
+        exec(Arisa.Modules.Attachment) { issue ->
+            "Attachment" to attachmentModule(
+                AttachmentModule.Request(
+                    issue.attachments
+                        .map { a ->
+                            AttachmentModule.Attachment(
+                                a.fileName,
+                                ::deleteAttachment.partially1(jiraClient).partially1(a)
+                            )
+                        }
+                )
+            )
+        }
+        exec(Arisa.Modules.CHK) { issue ->
+            "CHK" to chkModule(
+                CHKModule.Request(
+                    issue.getField(config[Arisa.CustomFields.chkField]) as? String?,
+                    ((issue.getField(config[Arisa.CustomFields.confirmationField])) as? JSONObject)?.get("value") as? String?,
+                    ::updateCHK.partially1(issue).partially1(config[Arisa.CustomFields.chkField])
+                )
+            )
+        }
+        exec(Arisa.Modules.Crash) { issue ->
+            "Crash" to crashModule(
+                CrashModule.Request(
+                    issue.attachments
+                        .map { a -> CrashModule.Attachment(a.fileName, a.createdDate, a.download()) },
+                    issue.description,
+                    issue.createdDate,
+                    ::resolveAs.partially1(issue).partially1("Invalid"),
+                    ::resolveAs.partially1(issue).partially1("Duplicate"),
+                    ::link.partially1(issue).partially1("Duplicate"),
+                    ::addComment.partially1(issue).partially1(config[Arisa.Modules.Crash.moddedMessage]),
+                    { key -> addComment(issue, config[Arisa.Modules.Crash.duplicateMessage].replace("{DUPLICATE}", key)) }
+                )
+            )
+        }
+        exec(Arisa.Modules.Empty) { issue ->
+            "Empty" to emptyModule(
+                EmptyModule.Request(
+                    issue.attachments.size,
+                    issue.description,
+                    issue.getField("environment") as? String?,
```

Comment:
[minor] We do the issue.getField and converted to string a lot, might make sense to have a fun Issue.getFieldAsString("fieldName") that does this cast to nullable string automatically